llvm-external-projects/iree-dialects/lib/Dialect/IREEPyDM/Transforms/ToIREE/LoweringPatterns.cpp

```diff
@@ -271,60 +304,173 @@ class ApplyCompareNumericConversion
   }
 };
 
-class BoolToPredConversion : public OpConversionPattern<pydm_d::BoolToPredOp> {
+class AssignSubscriptListConversion
+    : public OpConversionPattern<PYDM::AssignSubscriptOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      PYDM::AssignSubscriptOp srcOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto pySequence = srcOp.lhs();
+    if (!pySequence.getType().isa<PYDM::ListType>())
+      return rewriter.notifyMatchFailure(srcOp, "not builtin sequence");
+    auto pySlice = srcOp.slice();
+    if (!pySlice.getType().isa<PYDM::IntegerType>())
+      return rewriter.notifyMatchFailure(srcOp,
+                                         "slice is not static integer type");
+
+    auto loc = srcOp.getLoc();
+    auto sequence = adaptor.lhs();
+    auto slice = adaptor.slice();
+    auto indexType = rewriter.getType<IndexType>();
+    Type statusType =
+        getTypeConverter()->convertType(srcOp.exc_result().getType());
+    Value valueToSet =
+        boxIfNecessary(loc, pySequence.getType().cast<PYDM::ListType>(),
+                       srcOp.rhs().getType(), adaptor.rhs(), rewriter);
+    if (!valueToSet) {
+      return rewriter.notifyMatchFailure(
+          srcOp, "unsupported list assignment boxing mode");
+    }
+
+    Value zero = rewriter.create<arith::ConstantIntOp>(loc, 0, slice.getType());
```

Comment:
is this something you could use scf.if for? or would that not help? (I still want a real scf.switch that has regions)

Reply:
Worth resolving that a bit more globally. Right now, these programs are heavily/exclusively CFG based, and that makes a number of things about the type inference more true to the literature and priors. I'd like to avoid going into SCF land piecemeal, and when I've tried it adds a lot of complexity without a lot of benefit (and makes fine grained multi-versioning of paths through the CFG really hard).